lusidtools/cocoon/seed_sample_data.py

```diff
@@ -0,0 +1,89 @@
+import pandas as pd
+from lusidtools.cocoon.cocoon import load_from_data_frame
+from lusidtools.cocoon.utilities import load_json_file
+import logging
+
+logger = logging.getLogger()
+
+mappings = dict(load_json_file("config/seed_sample_data.json"))
+
+
+def seed_data(
+    api_factory,
+    domains,
+    scope: str,
+    transaction_file: str,
+    file_type: str,
+    mappings: dict = mappings,
+    sub_holding_keys=[],
+):
+
+    """
+    Description:
+    This function allows users to seed their LUSID environment with some core data (e.g. instruments,
+    portfolios, and transaction) from one file.
+
+    :param lusid.utilities.ApiClientFactory api_factory: The api factory to use.
+    :param list[str] domains: A list of the file_types for upload.
+    :param str scope: The scope of the transaction portfolio.
+    :param str transaction_file: a source file of transaction data.
+    :param str file_type: the file extension (e.g. "csv" for "test_transaction.csv".
+    :param dict mappings: a file containing mapping of DataFrame headers to LUSID headers.
+    :param list sub_holding_keys: a list of sub-holding keys for grouping.
+    """
+
+    # Gather a dictionary of supported files
+
+    supported_files = {"csv": "csv", "xlsx": "excel"}
+
+    # Check that the extension on the transction file is consistent with file_type argument
+
+    if str(transaction_file).lower().split(".")[-1] != file_type.lower():
```

Comment:
I think there's a `Path.suffix` you can use, also probably need to check the file exists

Reply:
Good idea. Updated.
Also added tests to check file exists condition works correctly when we pass a bad file.